tests/microbenchmarks/all_reduce.py

```
@@ -27,15 +29,101 @@
+            rtol=1e-2,
+            msg="Allreduce result mismatch",
+        )
+    return median_ms
+
+
+def allreduce_benchmark(
+    dtype: str = 'bfloat16',
+    test_range: str = "256,256000000,10",
+    enable_cudagraph: bool = False,
+    explore_2d: bool = False,
+    save_csv: str = None,
```

Ruff (0.14.0)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)